src/managers/guildChannels.ts

```diff
@@ -116,4 +116,13 @@ export class GuildChannelsManager extends BaseChildManager<
     )
     return this
   }
+
+  async keys(): Promise<string[]> {
+
+    const raw: GuildChannelPayload[] = await this.client.cache.array('channels') ?? []
+
+    if(raw.length === 0) return []
+
+    return raw.filter(rawChannel => rawChannel.guild_id === this.guild.id).map(ch => ch.id)
```

Comment:
This makes two different array allocations and is O(n^2). You should use an explicitly defined array and use for...of loop then push the IDs of channels which have right guild ID.